session/session_test.go

```diff
@@ -48,11 +52,51 @@ func TestReceiveEventWhenSubscribing(t *testing.T) {
 		if event != SessionDisconnected {
 			t.Error("Expected to receive disconnected: ", event)
 		}
+	case <-time.After(5 * time.Second):
+		t.Error("Failed to receive event")
+	}
 
-		if <-events != SessionReconnected {
+	select {
+	case event := <-events:
+		if event != SessionReconnected {
 			t.Error("Expected to receive reconnected: ", event)
 		}
-	case <-time.After(100 * time.Millisecond):
+	case <-time.After(5 * time.Second):
 		t.Error("Failed to receive event")
 	}
 }
+
+func TestResumeZKSessionWithValidSession(t *testing.T) {
+	proxy := Setup(t)
+	defer proxy.Delete()
+
+	clientId, err := zookeeper.LoadClientId(make([]byte, 24))
```

Comment:
Right now you're relying on the (I presume) undefined behaviour where `ResumeZKSession` just works when you pass it a clientid that is all zeroes. I would change this to get the client ID from an established session, call `clientId.Save()` to save it to a byte slice, then pass that to `LoadClientId`.